client/components/ProductDetail/ProductDetail.jsx

```diff
@@ -13,159 +13,80 @@ import ShareButtons from './ShareButtons.jsx';
 import ProductDescription from './ProductDescription.jsx';
 import ScrollToReviews from './ScrollToReviews.jsx';
 
-class ProductDetail extends Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      currentProduct: null,
-      productStyles: null,
-      currentStyle: null,
-      currentPhoto: null,
-      selectedSize: '',
-      selectedQuantity: '',
-      cart: [],
-      rating: null,
-      allRatings: null,
-    };
-
-    this.changeStyle = this.changeStyle.bind(this);
-    this.changePhoto = this.changePhoto.bind(this);
-    this.changeSize = this.changeSize.bind(this);
-    this.changeQuantity = this.changeQuantity.bind(this);
-    this.addToCart = this.addToCart.bind(this);
-  }
-
-  addToCart(id, size, quantity) {
-    var newCart = this.state.cart.slice();
-    var item = {
-      item: this.state.currentProduct,
-      style: this.state.currentStyle,
-      size: this.state.selectedSize.size,
-      quantity: this.state.selectedQuantity,
-    };
-    newCart.push(item);
-    this.setState({
-      cart: newCart,
-    });
-  }
-
-  changeSize(size) {
-    if (size === 'Select Size') {
-      this.setState({
-        selectedSize: '',
-      });
-    }
-    for (var key in this.state.currentStyle.skus) {
-      if (this.state.currentStyle.skus[key].size === size) {
-        this.setState({
-          selectedSize: this.state.currentStyle.skus[key],
-        });
-        break;
-      }
-    }
-    console.log(this.state.selectedSize);
-  }
+const ProductDetail = ({ product, productReviews, addToCart }) => {
+  console.log('>>>>>', product);
 
-  changeQuantity(quantity) {
-    this.setState({
-      selectedQuantity: quantity,
-    });
-  }
+  const [style, setStyle] = useState(product.styles.length > 0 ? product.styles[0] : null);
+  const [photo, setPhoto] = useState(
+    product.styles.length > 0 ? product.styles[0].photos[0] : null
```

Comment:
checking for `product.styles.length` wont help you if `product.styles.photos` is `undefined`. consider aliasing what you want in a variable up on line 18